### PR TITLE
report: tweak summary marker styles

### PR DESCRIPTION
**Pull request by connorjclark**
related: #12089

image

Latest Chrome decides to emit a deprecation warning for `::-webkit-details-marker` (it probably shouldn't...). this doesn't actually avoid that, but I thought we should use the new `::marker` just cuz.

still works in FF/Safari/Chrome. actually, nothing really changes :) this will be eventually necessary, I guess Chrome will remove old support one day?

### CODE REVIEW

File: lighthouse-core/report/html/report-styles.css [hunk 1]
```diff
@@ -541,8 +541,19 @@
...
+@supports selector(::marker) {
+  .lh-audit-group > summary,
+  .lh-expandable-details > summary {
+    display: block;
+  }
+}
+
+/* Older webkit browsers might have -webkit-details-marker,
+   and require setting list style type to none. */
 .lh-audit-group > summary,
```
Comment (connorjclark) [inline lighthouse-core/report/html/report-styles.css:557]: would have used `supports not selector(...)` but safari doesn't support that.

File: lighthouse-core/report/html/report-styles.css [hunk 2]
```diff
@@ -541,8 +541,19 @@
   background-color: var(--color-hover);
 }
 
-/* Hide the expandable arrow icon, three ways: via the CSS Counter Styles spec, for webkit/blink browsers, hiding the polyfilled icon */
```
Comment (connorjclark) [inline lighthouse-core/report/html/report-styles.css:544]: link is dead, i found the old sass code but it didn't seem helpful to have here

comment confused me so I just removed. maybe someone could help me reword? we aren't use @counter-styles (whatever that is), and I'm don't know what "polyfilled icon" is referring to.

File: lighthouse-core/report/html/report-styles.css [hunk 3]
```diff
@@ -541,8 +541,19 @@
...
-/* https://github.com/javan/details-element-polyfill/blob/master/src/details-element-polyfill/polyfill.sass */
+/* We want to hide the browser's default arrow marker on summary elements. */
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css:544]:
```css
/* We want to hide the browser's default arrow marker on summary elements. Admittedly, it's complicated. */
.lh-audit-group > summary,
.lh-expandable-details > summary {
  /* Blink 89+ will hide the arrow when display is changed from (new) default of `list-item` to block.  https://chromestatus.com/feature/6730096436051968*/
  display: block;
}
/* Safari, Firefox (surprisingly), and Blink <=88 require using the -webkit-details-marker selector */
.lh-audit-group > summary::-webkit-details-marker,
.lh-expandable-details > summary::-webkit-details-marker {
  display: none;
}
/* When the details-element-polyfill is used, it adds an ugly arrow. Hide it. https://github.com/javan/details-element-polyfill/blob/main/src/styles.js */
.lh-audit-group > summary:before,
.lh-expandable-details > summary:before {
  display: none;
}
```

File: lighthouse-core/report/html/report-styles.css (hunk 2 above)
Comment (paulirish) [reply lighthouse-core/report/html/report-styles.css:544]: counter styles spec was .... incorrect. 

"polyfilled icon" means the CSS added by deatils-element-polyfill (which is used in PSI for... old edge users and IE11 users?)  
we can remove that polyfill now, tbh.

**Review by paulirish — CHANGES_REQUESTED** [review]
thx for looking into this. i think we can siplify it even more.


and yeah thanks for filing the chromium issue.. agree it's impractical to not use the -webkit prefix if you care about either safari or FF.

File: lighthouse-core/report/html/report-styles.css (hunk 3 above)
Comment (connorjclark) [reply lighthouse-core/report/html/report-styles.css:544]: I tweaked my comments a bit, but IMO it was already specifying why things were being done at an appropriate level of detail (new/old browsers, tale as old as time). The provided links provides more detail (like what version of blink the newer stuff is supported).

got rid of the `@supports` block too, the idea there was to eventually have chrome not emit a deprecation warning if it was smart and checked for these rules in a better way; but the deprecation is now removed, so whatever.

Comment (paulirish) [reply lighthouse-core/report/html/report-styles.css:544]: I'm unclear why this has to be contentious.. 

> I tweaked my comments a bit, but IMO it was already specifying why things were being done at an appropriate level of detail

if that were the case, then you would have deleted the `list-style-type: none;` rule. it's totally unnecessary.

**Review by paulirish — APPROVED** [review]

**Pull request merged**